File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/entity/DbDomainRecord.java
package com.platdmit.simplecloudmanager.data.database.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(foreignKeys = @ForeignKey(entity = DbDomain.class, parentColumns = "mId", childColumns = "mParentDomain"))
public class DbDomainRecord {

    @PrimaryKey
    private long mId;
    private String mPriority;
    private String mPort;
    private String mWeight;
    private String mTtl;

    private String mType;
    private String mName;
    private String mData;

    @ColumnInfo(index = true)
    private long mParentDomain;

    public DbDomainRecord(long id, String priority, String port, String weight, String ttl, String type, String name, String data, long parentDomain) {
        mId = id;
        mPriority = priority;
        mPort = port;
        mWeight = weight;
        mTtl = ttl;
        mType = type;
        mName = name;
        mData = data;
        mParentDomain = parentDomain;
    }

    public long getId() {
        return mId;
    }

    public String getPriority() {
        return mPriority;
    }

    public String getPort() {
        return mPort;
    }

    public String getWeight() {
        return mWeight;
    }

    public String getTtl() {
        return mTtl;
    }

    public String getType() {
        return mType;
    }

    public String getName() {
        return mName;
    }

    public String getData() {
        return mData;
    }

    public long getParentDomain() {
        return mParentDomain;
    }
}

File: data/src/main/java/com/platdmit/data/room/entity/DbDomainRecord.kt
package com.platdmit.data.room.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(foreignKeys = [ForeignKey(entity = DbDomain::class, parentColumns = ["id"], childColumns = ["parentDomain"])])
data class DbDomainRecord(
        @PrimaryKey val id: Long,
        val priority: String,
        val port: String,
        val weight: String,
        val ttl: String,
        val type: String,
        val name: String,
        val data: String,
        @ColumnInfo(index = true) val parentDomain: Long
)